client/src/main/scala/skuber/json/package.scala

```diff
@@ -472,23 +472,22 @@ package object format {
   implicit val lifecycleFormat: Format[Lifecycle] = Json.format[Lifecycle]
   
   import Volume._
-  
-  implicit val emptyDirReads: Reads[EmptyDir] = {
-      (JsPath \ "medium").readNullable[String].map {
-        case Some(med) if med == "Memory" => EmptyDir(MemoryStorageMedium)
-        case Some(med) if med == "HugePages" => EmptyDir(HugePagesStorageMedium)
-        case _ => EmptyDir(DefaultStorageMedium)
-      }
-  }  
-  implicit val emptyDirWrites: Writes[EmptyDir] = Writes[EmptyDir] {
-    ed => ed.medium match {
-      case DefaultStorageMedium => (JsPath \ "medium").write[String].writes("")
-      case MemoryStorageMedium => (JsPath \ "medium").write[String].writes("Memory")
-      case HugePagesStorageMedium => (JsPath \ "medium").write[String].writes("HugePages")
-    }
-  }  
-  implicit val emptyDirFormat: Format[EmptyDir] = Format(emptyDirReads, emptyDirWrites)
-  
+
+  implicit val storageMediumFormat: Format[StorageMedium] = Format[StorageMedium](Reads[StorageMedium] {
+    case JsString(med) if med == "Memory" => JsSuccess(MemoryStorageMedium)
+    case JsString(med) if med == "HugePages" => JsSuccess(HugePagesStorageMedium)
+    case _ => JsSuccess(DefaultStorageMedium)
+  }, Writes[StorageMedium] {
+    case DefaultStorageMedium => JsString("")
+    case MemoryStorageMedium => JsString("Memory")
+    case HugePagesStorageMedium => JsString("HugePages")
+  })
+
+  implicit val emptyDirFormat: Format[EmptyDir] = (
+    (JsPath \ "medium").format[StorageMedium] and
```

Comment:
I think  `medium` is an optional field, but this will expect it to be there so would error on reading it if it is omitted.
Best way to test that is to have some explicit json with an empty `emptyDir` in the `VolumeFormatSpec` test below, something equivalent to e.g. the volumes spec in the example pod [here](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)